include/dsn/cpp/rpc_stream.h

```diff
@@ -97,6 +98,9 @@ namespace dsn
         {
             _last_write_next_committed = true;
             _last_write_next_total_size = 0;
+#ifdef DSN_USE_THRIFT_SERIALIZATION
+            _p_value_id = &(reinterpret_cast<dsn::message_ex*>(msg)->_value_id);
```

Comment:
Are these specially added parameters for message parsing etc.? If yes, can we put them into the message parser instead? As one message parser is used for one  In that case, things will be much simpler.